build.rb

```diff
@@ -449,10 +457,17 @@ def extract_all()
 def pack_all(skip_compile=false)
   verify_executables
   init_directories()
-  should_compile = update_gffs() && !skip_compile
-  puts "[INFO] No change in nss sources detected. Skipping compilation." unless should_compile && !skip_compile
-  puts "[INFO] Skipping compilation due to configuration" if skip_compile
-  compile_nss(MODULE_FILE) if should_compile
+
+  if skip_compile
+    puts "[INFO] Skipping compilation due to configuration"
+  end
+    
+  should_compile = update_gffs()
+  if should_compile && !skip_compile
+    puts "[INFO] No change in nss sources detected. Skipping compilation."
+  end
+
+  compile_nss(MODULE_FILE) unless skip_compile
```

Comment:
The logic above doesn't look right. This function was a mess already, and I think it needs to be redone in a more sensible way. The two flags make it complex, and there's something not right about conditionals that log an action without actually doing anything.

I would suggest replacing L461-468 with
```ruby
  nss_changed = update_gffs()
  should_compile = nss_changed && !skip_compile
```
And this line to
```suggestion
  if should_compile
    compile_nss(MODULE_FILE)
  else
    if skip_compile
      puts "[INFO] Skipping compilation due to configuration"
    else
      puts "[INFO] No change in nss sources detected. Skipping compilation."
    end
  end
```
Or perhaps you see something more logical and/or compact.